In [ ]:
# %pip install pandas openpyxl 
#   use of excel writer

In [2]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

import pingouin as pg

In [ ]:
#   File Beer Profiles and ratings.csv
#   --------------------------------------

# This file contains 1 Tabs  with Beer dateset detailled profile and ratings info
    #   Objective of this script - check relations between dimension concepts


In [3]:
# Fetch raw data

rawdata = pd.read_csv(r'C:\Users\t.vincotte\Documents\CODE\TFE\EoF - Work\Beer_Profile\Source_Files\beer_profile_and_ratings.csv') 
 

In [4]:
rawdata.count()

Name                 3197
Style                3197
Brewery              3197
Beer Name (Full)     3197
Description          3197
ABV                  3197
Min IBU              3197
Max IBU              3197
Astringency          3197
Body                 3197
Alcohol              3197
Bitter               3197
Sweet                3197
Sour                 3197
Salty                3197
Fruits               3197
Hoppy                3197
Spices               3197
Malty                3197
review_aroma         3197
review_appearance    3197
review_palate        3197
review_taste         3197
review_overall       3197
number_of_reviews    3197
dtype: int64

In [5]:
#rawdata.info()
str = rawdata.info()
structure = pd.DataFrame(str)  #  ne met pas en dataframe
structure
# structure.to_excel("Beer_datasets_structures",sheet_name='Beer data set details') # n ecrit pas le rawdata.info dans le fichier

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3197 entries, 0 to 3196
Data columns (total 25 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Name               3197 non-null   object 
 1   Style              3197 non-null   object 
 2   Brewery            3197 non-null   object 
 3   Beer Name (Full)   3197 non-null   object 
 4   Description        3197 non-null   object 
 5   ABV                3197 non-null   float64
 6   Min IBU            3197 non-null   int64  
 7   Max IBU            3197 non-null   int64  
 8   Astringency        3197 non-null   int64  
 9   Body               3197 non-null   int64  
 10  Alcohol            3197 non-null   int64  
 11  Bitter             3197 non-null   int64  
 12  Sweet              3197 non-null   int64  
 13  Sour               3197 non-null   int64  
 14  Salty              3197 non-null   int64  
 15  Fruits             3197 non-null   int64  
 16  Hoppy              3197 

Empty DataFrame
Columns: []
Index: []

In [6]:
rawdata.head(1)

Name    Style              Brewery                   Beer Name (Full)  \
0  Amber  Altbier  Alaskan Brewing Co.  Alaskan Brewing Co. Alaskan Amber   

                                         Description  ABV  Min IBU  Max IBU  \
0  Notes:Richly malty and long on the palate, wit...  5.3       25       50   

   Astringency  Body  ...  Fruits  Hoppy  Spices  Malty  review_aroma  \
0           13    32  ...      33     57       8    111      3.498994   

   review_appearance  review_palate  review_taste  review_overall  \
0           3.636821       3.556338      3.643863        3.847082   

   number_of_reviews  
0                497  

[1 rows x 25 columns]

In [ ]:
#   Only show columns we want to see - methods not used here
#   ----------------------------------------------------------

        # Using DataFrame.drop
            #   df.drop(df.columns[[1, 2]], axis=1, inplace=True)

        # Select the ones you want
            #   df1 = df[['a','d']]


In [8]:
#   Beer dimension check - rough listing - create spreadsheet 
#   ----------------------------------------------------------

    #   https://www.geeksforgeeks.org/exporting-a-pandas-dataframe-to-an-excel-file/
    #   https://stackoverflow.com/questions/42370977/how-to-save-a-new-sheet-in-an-existing-excel-file-using-pandas
    #   https://saturncloud.io/blog/how-to-add-a-worksheet-to-an-existing-excel-file-with-pandas/ 

df1 = rawdata.drop(['Description'], axis=1)
df1.to_excel("EDA_DimCheck_BeerProfileRating.xlsx",sheet_name='rough_listing')

        #   Note : (Re)creates spreadsheet as running - below - use method to append sheet


Alternate method (stack overflow) - Add sheet to existing file
-------------------------------------
import pandas as pd
import numpy as np

path = r"C:\Users\fedel\Desktop\excelData\PhD_data.xlsx"

x3 = np.random.randn(100, 2)
df3 = pd.DataFrame(x3)

x4 = np.random.randn(100, 2)
df4 = pd.DataFrame(x4)

writer = pd.ExcelWriter(path, engine = 'xlsxwriter')
df3.to_excel(writer, sheet_name = 'x3')
df4.to_excel(writer, sheet_name = 'x4')
writer.save()
writer.close()

----

import pandas as pd

data = {'EmployeeID': [101, 102, 103],
        'EmployeeName': ['Alice', 'Bob', 'Charlie'],
        'Salary': [60000, 70000, 80000]}

df_new = pd.DataFrame(data)

with pd.ExcelWriter('existing_file.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df_new.to_excel(writer, sheet_name='New Sheet', index=False)

In [9]:
#   Beer dimension check - List by style
#   ---------------------------------------
    #   Use method to append sheet to existing excel file

df1_by_style = df1[['Style','Name','Brewery','ABV']].sort_values(by=['Name']).sort_values(by=['Style'])
with pd.ExcelWriter('EDA_DimCheck_BeerProfileRating.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_style.to_excel(writer, sheet_name='list_by_style', index=True)

# df1_by_style.to_excel("EDA_DimCheck_BeerDataset.xlsx",sheet_name='list_by_style')


In [10]:
#   Beer dimension check - List by brewery
#   ---------------------------------------
   #   Use method to append sheet to existing excel file

df1_by_brewery = df1[['Brewery','Name','Style','ABV']].sort_values(by=['Name']).sort_values(by=['Brewery'])
with pd.ExcelWriter('EDA_DimCheck_BeerProfileRating.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_brewery.to_excel(writer, sheet_name='list_by_brewery', index=True)


In [11]:
#   Beer dimension check - List by style_brewery
#   ---------------------------------------------
   #   Use method to append sheet to existing excel file

df1_by_brewery_style = df1[['Style','Brewery','Name','ABV']].sort_values(by=['Name']).sort_values(by=['Brewery']).sort_values(by=['Style'])
with pd.ExcelWriter('EDA_DimCheck_BeerProfileRating.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_brewery_style.to_excel(writer, sheet_name='list_by_brewery_style', index=True)


In [12]:
#   Beer dimension check - List by brewery_style
#   ---------------------------------------------
   #   Use method to append sheet to existing excel file

df1_by_brewery_style = df1[['Brewery','Style','Name','ABV']].sort_values(by=['Name']).sort_values(by=['Style']).sort_values(by=['Brewery'])
with pd.ExcelWriter('EDA_DimCheck_BeerProfileRating.xlsx', engine='openpyxl', mode='a') as writer:
    # Write the new DataFrame to a new sheet
    df1_by_brewery_style.to_excel(writer, sheet_name='list_by_style_brewery', index=True)


In [13]:
#     Check if dupplicate beer names
#   -------------------------------------
#   
df1_duplicate_beer_check = df1[['Name','ABV']].count()
df1_duplicate_beer_check 


Name    3197
ABV     3197
dtype: int64

In [14]:
#     Check if dupplicate beer names
#   -------------------------------------
#   
 
df2x = df1[['Name']].drop_duplicates
#df2 = pd.DataFrame[df2x]
df2x

#df2_duplicate_beer_check = df2.count()
#df2_duplicate_beer_check 

<bound method DataFrame.drop_duplicates of                                           Name
0                                        Amber
1                                   Double Bag
2                               Long Trail Ale
3                                 Doppelsticke
4                 Sleigh'r Dark Doüble Alt Ale
...                                        ...
3192                           Winter Shredder
3193                        The First Snow Ale
3194                       Red Nose Winter Ale
3195                      Fish Tale Winterfish
3196  Frosted Frog Christmas Ale - Barrel-Aged

[3197 rows x 1 columns]>

Afre drop dupplicate still amount of rows >> no duplicate in beer name

3197 records